app/controllers/buyers/applications_controller.rb

```diff
@@ -52,25 +55,28 @@ def index
   end
 
   def new
-    @cinstance = @buyer.bought_cinstances.build
-    extend_cinstance_for_new_plan
-    @plans = accessible_plans.stock
+    @products = accessible_services
 
     if params[:account_id]
-      @account = current_account.buyers.find params[:account_id]
-      activate_menu :buyers, :accounts
+      # We're under Account context
+      find_buyer
+      @cinstance = @account.bought_cinstances.build
+      extend_cinstance_for_new_plan
+      activate_menu :buyers, :accounts, :listing
+    else
+      # We're under Applications context
+      activate_menu :audience, :applications, :listing
     end
```

Comment:
I was thinking about making this more readable. WDYT?

```ruby
  def create
    @products = accessible_services

    return activate_menu :audience, :applications, :listing if applications_context?

    find_buyer
    @cinstance = @account.bought_cinstances.build
    extend_cinstance_for_new_plan

    activate_menu :buyers, :accounts, :listing
  end

  protected

  def buyers_context?
    params[:account_id]
  end

  def applications_context?
    !buyers_context?
  end
```

Reply:
I don't think we need both context mehods though